force-app/main/default/aura/stratcraft/stratcraftController.js

```diff
@@ -3,96 +3,69 @@
 		helper.initTree(component, event, helper);
 	},
 
-	 // testing future logic for creating components
-	handleCreateComponents : function(component, event, helper) { 
-		 $A.createComponents(
-        	[
-			    ["ui:inputText",{
-			    	"name" : "test1",
-			    	"label" : "test3",
-			        "value" : "12345"
-			    }],
-			    ["ui:inputText",{
-			    	"name" : "test2",
-			    	"label" : "test4",
-			        "value" : "1234"
-			    }]
-		    ],
-            function(components, status, errorMessage){
-                //Add the new button to the body array
-                if (status === "SUCCESS") {
-					var body = component.get("v.body");
-                    components.forEach(function(entry) {
-                        body.push(entry);
-                    });                    
-                    
-                    component.set("v.body", body);
-                }
-                else if (status === "INCOMPLETE") {
-                    console.log("No response from server or client is offline.")
-                    // Show offline error
-                }
-                else if (status === "ERROR") {
-                    console.log("Error: " + errorMessage);
-                    // Show error message
-                }
-            }
-        );
-	},
-
     handleUploadFinished: function (cmp, event) {
         // Get the list of uploaded files
         var uploadedFiles = event.getParam("files");
         alert("Files uploaded : " + uploadedFiles.length);
     },
 
-    onDrop: function(component, event) { 
+    onDrop: function(cmp, event, helper) { 
 		event.stopPropagation(); 
 		event.preventDefault(); 
-
+		var reader = new FileReader();
 		var files = event.dataTransfer.files; 
+	    var spinner = cmp.find("mySpinner");
+      	$A.util.toggleClass(spinner, "slds-hide");
 		for (var i = 0; i < files.length; i++) { 
 			var file = files[i]; 
-			var reader = new FileReader(); 
-		
+			reader = new FileReader(); 		
 			reader.onloadend = function() { 
 				console.log("uploaded file data is: " + reader.result); 		
-				component.set("v.strategyXML", reader.result);
+				cmp.set("v.strategyXML", reader.result);	
+				var cmpEvent = cmp.getEvent("setStrategy");
+		        cmpEvent.fire();
 			}; 
-
 			reader.readAsText(file); 
-		} 
+		}
 	}, 
 
-	handleClick: function (cmp, event, helper) {
```

Comment:
Let's call this LoadStrategy instead of SetStrategy. "Set" has some other meanings that confuse the issue a little here.